### PR TITLE
Convert `PR-Trigger/index.js` to TS

### PR DESCRIPTION
**Pull request by elibarzilay**
* Removed `sideEffects` and instead just try it for the single entry
  that was there.

* Refactored things a bit around `acceptedEventsToActions` and `allowListedActions`

* Dropped `if (state.type === "noop")` (similar to #294)

* Some touchups in `src/side-effects/merge-codeowner-prs`

### CODE REVIEW

File: PR-Trigger/index.ts [hunk 1]
```diff
@@ -37,34 +34,31 @@ const httpTrigger = async function (context, _req) {
 
     // Allow the bot to run side-effects which are not the 'core' function
     // of the review cycle, but are related to keeping DT running smoothly
-    const sideEffects = {
-        "check_suite": mergeCodeOwnersOnGreen
+    if (event === "check_suite") {
+        await mergeCodeOwnersOnGreen(req.body as EventPayloads.WebhookPayloadCheckSuite);
```
Comment (orta) [inline PR-Trigger/index.ts:38]: Good call on some of these simplifications - now we know what this codebase is, we don't really need it

File: PR-Trigger/index.ts [hunk 2]
```diff
@@ -37,34 +34,31 @@ const httpTrigger = async function (context, _req) {
...
-        "check_suite": mergeCodeOwnersOnGreen
+    if (event === "check_suite") {
+        await mergeCodeOwnersOnGreen(req.body as EventPayloads.WebhookPayloadCheckSuite);
     }
 
-    if (sideEffects[event]) sideEffects[event](req.body);
-
     // https://developer.github.com/webhooks/
-    const acceptedEventsToActions = {
+    const acceptedEventsToActions: Record<string, string[] | undefined> = {
```
Comment (orta) [inline PR-Trigger/index.ts:42]: Heh, you can always turn on http://typescriptlang.org/tsconfig#noUncheckedIndexedAccess

**Review by orta — APPROVED** [review]
Looks good to me

File: PR-Trigger/index.ts (hunk 1 above)
Comment (elibarzilay) [reply PR-Trigger/index.ts:38]: 👍

**Pull request merged**